awx/main/models/jobs.py

```diff
@@ -291,14 +297,13 @@ def validation_errors(self):
                 _("Job Template must provide 'inventory' or allow prompting for it."),
             ]
         if self.project is None:
-            validation_errors['project'] = [
-                _("Job Templates must have a project assigned."),
-            ]
+            if self.fqcn_role == '' or self.fqcn_role is None:
+                validation_errors['project'] = [_("Job Templates must have a role or project assigned")]
         return validation_errors
 
     @property
     def resources_needed_to_start(self):
-        return [fd for fd in ['project', 'inventory'] if not getattr(self, '{}_id'.format(fd))]
+        return [fd for fd in ['inventory'] if not getattr(self, '{}_id'.format(fd))]
```

Comment:
This is a mostly unacceptable change, but it gets us past the check itself. There's a complex layer of validation on these properties and it feels like we need another representation. At least at this level.